Failed to decode image for window icon

### Describe the bug:
Flakey test in Fyne test suite:
```
2020/03/29 13:33:08 Fyne error:  Failed to decode image for window icon
2020/03/29 13:33:08   Cause: image: unknown format
2020/03/29 13:33:08   At: /Users/stuartscott/Documents/Projects/Go/src/fyne.io/fyne/internal/driver/glfw/window.go:215
--- FAIL: TestWindow_Focus (0.13s)
    window_test.go:726: 
        	Error Trace:	window_test.go:726
        	Error:      	Not equal: 
        	            	expected: "abcd"
        	            	actual  : "bcd"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-abcd
        	            	+bcd
        	Test:       	TestWindow_Focus
FAIL
FAIL	fyne.io/fyne/internal/driver/glfw	5.338s
```

### To Reproduce:
Steps to reproduce the behaviour:
1. go test -count=1 fyne.io/fyne/... (may need to run tests multiple times)

### Screenshots:
N/A

### Example code:
N/A

### Device (please complete the following information):
 - OS: MacOS
 - Version: 10.15.4 (Catalina)
 - Go version: go1.13.8 darwin/amd64
 - Fyne version: develop/a85639c2c5ffba3cbaf962a156b5961b3f3349d4

--- a/internal/driver/glfw/menu_darwin_test.go
+++ b/internal/driver/glfw/menu_darwin_test.go
@@ -131,7 +131,7 @@ func resetMainMenu() {
 	j := 0
 	for i := 0; i < testNSMenuNumberOfItems(mainMenu); i++ {
 		menu := testNSMenuItemSubmenu(testNSMenuItemAtIndex(mainMenu, i))
-		if testNSMenuTitle(menu) == initialMenus[j] {
+		if j < len(initialMenus) && testNSMenuTitle(menu) == initialMenus[j] {
 			j++
 			continue
 		}

--- a/internal/driver/glfw/window.go
+++ b/internal/driver/glfw/window.go
@@ -208,6 +208,11 @@ func (w *window) SetIcon(icon fyne.Resource) {
 		return
 	}
 
+	if string(icon.Content()[:4]) == "<svg" {
+		fyne.LogError("Window icon does not support vector images", nil)
+		return
+	}
+
 	w.runOnMainWhenCreated(func() {
 		if w.icon == nil {
 			w.viewport.SetIcon(nil)

--- a/internal/driver/glfw/window_test.go
+++ b/internal/driver/glfw/window_test.go
@@ -635,7 +635,7 @@ func TestWindow_SetIcon(t *testing.T) {
 	w := createWindow("Test")
 	assert.Equal(t, fyne.CurrentApp().Icon(), w.Icon())
 
-	newIcon := theme.CancelIcon()
+	newIcon := theme.FyneLogo()
 	w.SetIcon(newIcon)
 	assert.Equal(t, newIcon, w.Icon())
 }

